File: domain/usecases/get_development_packages.py
from typing import List
from domain.entities import Package
from data.repositories.appstream_repository import AppStreamFeaturedRepository

class GetDevelopmentPackagesUseCase:
    def __init__(self, appstream_repo: AppStreamFeaturedRepository):
        self.appstream_repo = appstream_repo

    def execute(self) -> List[Package]:
        return self.appstream_repo.get_development_packages()

File: src-rust/src/domain/usecases/get_development_packages.rs
use std::sync::Arc;
use crate::domain::{entities::Package, repositories::FeaturedRepository};

pub struct GetDevelopmentPackagesUseCase {
    pub appstream_repo: Arc<dyn FeaturedRepository>,
}
impl GetDevelopmentPackagesUseCase {
    pub fn new(appstream_repo: Arc<dyn FeaturedRepository>) -> Self {
        Self { appstream_repo }
    }
    pub async fn execute(&self) -> Vec<Package> {
        self.appstream_repo.get_development_packages().await
    }
}
